src/app/vendors/actions.ts

```diff
@@ -23,22 +39,149 @@ export function loadAllVendors(
   force = false,
 ): ThunkResult {
   return async (dispatch, getState) => {
-    // Only load at most once per 30 seconds
-    if (
-      !force &&
-      Date.now() -
-        (getState().vendors.vendorsByCharacter[characterId]?.lastLoaded?.getTime() || 0) <
-        30 * 1000
-    ) {
+    const timeSinceLastLoad =
+      Date.now() - (getState().vendors.vendorsByCharacter[characterId]?.lastLoaded || 0);
+
+    // Only load "all vendors" at most once per 30 seconds
+    if (!force && timeSinceLastLoad < 30 * 1000) {
       return;
     }
 
+    const timings = { salesTime: 0, componentsNeeded: 0, componentsFetched: 0, componentsTime: 0 };
+    let start = Date.now();
     try {
-      const vendorsResponse = await getVendorsApi(account, characterId);
+      // https://github.com/Bungie-net/api/issues/1887 MAY 2024-ish: for perf reasons,
+      // bungie has decided not to send back item components for the "all vendors" endpoint.
+      // this request still lets us know which items are sold, by all available vendors.
+      // enough to make a reasonable-looking vendors page full of shiny icons.
+      const vendorsResponse = await getVendors(account, characterId);
+      timings.salesTime = Date.now() - start;
       dispatch(loadedAll({ vendorsResponse, characterId }));
+
+      const defs = manifestSelector(getState());
+      if (
+        // we're done if this is d1. or if defs aren't loaded?
+        // can this kick off before defs are loaded?? what if it does???
```

Comment:
You might be able to put this in a store observable, so that it will kick off after defs are loaded?

Reply:
I think it always kicks off after defs are loaded.

Reply:
great to be reassured of, definitely appeared to that way in all testing